native/src/main/java/io/ballerina/stdlib/xmldata/utils/XmlDataUtils.java

```diff
@@ -62,7 +62,7 @@ public static BError getError(String message) {
     public static Object getModifiedRecord(BMap<BString, Object> input, BTypedesc type) {
         Type describingType = type.getDescribingType();
         Object value = input.get(input.getKeys()[0]);
-        if (value instanceof BArray) {
+        if (describingType.getTag() == TypeTags.MAP_TAG && value instanceof BArray) {
```

Comment:
why are we special-casing the first element?

Reply:
This implementation is for map<anydata[]>. We can't directly check whether the map value is an array or not. So, we have to check the value type. So, we have to get the value and check that type. Here, the first value's type is only checked. So, when the first field of the record is an array, that comes here.

Reply:
Let's create an issue and improve it as different task